File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/framework/socket/entity/request/SendMessageRequest.java
package ru.usedesk.chat_sdk.internal.data.framework.socket.entity.request;

public class SendMessageRequest extends BaseRequest {

    private static final String TYPE = "@@server/chat/SEND_MESSAGE";

    private final RequestMessage message;

    public SendMessageRequest(String token, RequestMessage requestMessage) {
        super(TYPE, token);
        this.message = requestMessage;
    }

    public RequestMessage getRequestMessage() {
        return message;
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/framework/socket/entity/request/SendMessageRequest.kt
package ru.usedesk.chat_sdk.internal.data.framework.socket.entity.request

import ru.usedesk.chat_sdk.internal.domain.entity.UsedeskFile

class SendMessageRequest(
        token: String,
        text: String? = " ",
        usedeskFile: UsedeskFile? = null
) : BaseRequest(TYPE, token) {

    private val message = RequestMessage(text, usedeskFile)

    companion object {
        private const val TYPE = "@@server/chat/SEND_MESSAGE"
    }

    private class RequestMessage(
            private val text: String?,
            private val usedeskFile: UsedeskFile?
    )
}